homeassistant/components/hydrawise/binary_sensor.py

```diff
@@ -24,13 +24,17 @@ class HydrawiseBinarySensorEntityDescription(BinarySensorEntityDescription):
     """Describes Hydrawise binary sensor."""
 
     value_fn: Callable[[HydrawiseBinarySensor], bool | None]
+    always_available: bool = False
 
 
 CONTROLLER_BINARY_SENSORS: tuple[HydrawiseBinarySensorEntityDescription, ...] = (
     HydrawiseBinarySensorEntityDescription(
         key="status",
         device_class=BinarySensorDeviceClass.CONNECTIVITY,
-        value_fn=lambda status_sensor: status_sensor.coordinator.last_update_success,
+        value_fn=lambda status_sensor: status_sensor.coordinator.last_update_success
+        and status_sensor.controller.online,
```

Comment:
Please use parenthesis around the lambda and indent it so it's easier to distinguish it from the description parameters.